File: app/src/main/java/com/example/myapplication/AssistantActivity.java
package com.example.myapplication;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import com.google.android.material.button.MaterialButton;
import com.google.android.material.card.MaterialCardView;
import com.google.android.material.chip.Chip;
import java.util.ArrayList;
import java.util.List;

public class AssistantActivity extends AppCompatActivity {

    private EditText edtAge, edtWeight, edtHeight;
    private Chip chipBackPain, chipNeckPain, chipStress, chipFatigue, chipInsomnia;
    private Chip chipRelaxation, chipPainRelief, chipRecovery, chipEnergy;
    private MaterialButton btnGetRecommendation, btnApplyRecommendation;
    private LinearLayout layoutLoading;
    private MaterialCardView cardRecommendation;
    private TextView txtRecommendedMode, txtRecommendedLevel, txtRecommendedDuration;
    private TextView txtRecommendedHeat, txtExplanation;

    private MassageRecommendation currentRecommendation;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_assistant);

        initViews();
        setupListeners();
    }

    private void initViews() {
        findViewById(R.id.btnBack).setOnClickListener(v -> finish());

        edtAge = findViewById(R.id.edtAge);
        edtWeight = findViewById(R.id.edtWeight);
        edtHeight = findViewById(R.id.edtHeight);

        chipBackPain = findViewById(R.id.chipBackPain);
        chipNeckPain = findViewById(R.id.chipNeckPain);
        chipStress = findViewById(R.id.chipStress);
        chipFatigue = findViewById(R.id.chipFatigue);
        chipInsomnia = findViewById(R.id.chipInsomnia);

        chipRelaxation = findViewById(R.id.chipRelaxation);
        chipPainRelief = findViewById(R.id.chipPainRelief);
        chipRecovery = findViewById(R.id.chipRecovery);
        chipEnergy = findViewById(R.id.chipEnergy);

        btnGetRecommendation = findViewById(R.id.btnGetRecommendation);
        btnApplyRecommendation = findViewById(R.id.btnApplyRecommendation);
        layoutLoading = findViewById(R.id.layoutLoading);
        cardRecommendation = findViewById(R.id.cardRecommendation);

        txtRecommendedMode = findViewById(R.id.txtRecommendedMode);
        txtRecommendedLevel = findViewById(R.id.txtRecommendedLevel);
        txtRecommendedDuration = findViewById(R.id.txtRecommendedDuration);
        txtRecommendedHeat = findViewById(R.id.txtRecommendedHeat);
        txtExplanation = findViewById(R.id.txtExplanation);
    }

    private void setupListeners() {
        btnGetRecommendation.setOnClickListener(v -> getAIRecommendation());
        btnApplyRecommendation.setOnClickListener(v -> applyRecommendation());

        // Quick presets
        findViewById(R.id.cardMorningEnergy).setOnClickListener(v ->
                applyPreset(new MassageRecommendation(4, 10, false, "Morning Energy",
                        "Quick energizing massage to start your day")));

        findViewById(R.id.cardRelaxation).setOnClickListener(v ->
                applyPreset(new MassageRecommendation(2, 20, true, "Evening Relaxation",
                        "Gentle massage with heat for deep relaxation")));

        findViewById(R.id.cardDeepTissue).setOnClickListener(v ->
                applyPreset(new MassageRecommendation(5, 15, true, "Deep Tissue",
                        "Intense massage for muscle recovery")));
    }



    private MassageRecommendation generateRecommendation(int age, float weight, float height,
                                                         List<String> conditions, String goal) {
        int level = 3; // Default medium
        int duration = 15;
        boolean heat = false;
        String mode = "Balanced Massage";
        StringBuilder explanation = new StringBuilder();

        // Calculate BMI
        float bmi = weight / ((height/100) * (height/100));

        // Age-based adjustments
        if (age < 25) {
            level = 4;
            duration = 15;
            explanation.append("Young age: higher intensity recommended. ");
        } else if (age < 40) {
            level = 3;
            duration = 15;
            explanation.append("Prime age: moderate intensity for balance. ");
        } else if (age < 60) {
            level = 2;
            duration = 20;
            heat = true;
            explanation.append("Middle age: gentle with heat therapy. ");
        } else {
            level = 1;
            duration = 15;
            heat = true;
            explanation.append("Senior: very gentle massage recommended. ");
        }

        // BMI adjustments
        if (bmi > 25) {
            duration += 5;
            explanation.append("Higher BMI: longer duration beneficial. ");
        }

        // Condition-based adjustments
        if (conditions.contains("back_pain") || conditions.contains("neck_pain")) {
            heat = true;
            if (level > 3) level = 3;
            explanation.append("Pain conditions: heat therapy + moderate intensity. ");
            mode = "Pain Relief Mode";
        }

        if (conditions.contains("stress") || conditions.contains("insomnia")) {
            level = Math.min(level, 2);
            duration = Math.max(duration, 20);
            heat = true;
            explanation.append("Stress/insomnia: gentle, long session with heat. ");
            mode = "Relaxation Mode";
        }

        if (conditions.contains("fatigue")) {
            level = Math.max(level, 3);
            explanation.append("Fatigue: moderate intensity to boost circulation. ");
        }

        // Goal-based adjustments
        switch (goal) {
            case "pain_relief":
                heat = true;
                level = Math.min(level, 3);
                mode = "Therapeutic Mode";
                break;
            case "recovery":
                level = Math.max(level, 3);
                duration = 20;
                mode = "Recovery Mode";
                explanation.append("Recovery goal: longer, intense session. ");
                break;
            case "energy":
                level = Math.max(level, 4);
                duration = 10;
                heat = false;
                mode = "Energy Boost Mode";
                explanation.append("Energy boost: short, intense session. ");
                break;
        }

        // Safety limits
        level = Math.max(1, Math.min(5, level));
        duration = Math.max(10, Math.min(30, duration));

        return new MassageRecommendation(level, duration, heat, mode, explanation.toString());
    }
    private void getAIRecommendation() {
        // Validate input
        if (edtAge.getText().toString().isEmpty() ||
                edtWeight.getText().toString().isEmpty() ||
                edtHeight.getText().toString().isEmpty()) {
            Toast.makeText(this, "Please fill in all fields", Toast.LENGTH_SHORT).show();
            return;
        }

        int age = Integer.parseInt(edtAge.getText().toString());
        float weight = Float.parseFloat(edtWeight.getText().toString());
        float height = Float.parseFloat(edtHeight.getText().toString());

        // Get health conditions
        List<String> conditions = new ArrayList<>();
        if (chipBackPain.isChecked()) conditions.add("back_pain");
        if (chipNeckPain.isChecked()) conditions.add("neck_pain");
        if (chipStress.isChecked()) conditions.add("stress");
        if (chipFatigue.isChecked()) conditions.add("fatigue");
        if (chipInsomnia.isChecked()) conditions.add("insomnia");

        // Get goal
        final String goal;
        if (chipPainRelief.isChecked()) {
            goal = "pain_relief";
        } else if (chipRecovery.isChecked()) {
            goal = "recovery";
        } else if (chipEnergy.isChecked()) {
            goal = "energy";
        } else {
            goal = "relaxation";
        }

        // Show loading
        layoutLoading.setVisibility(View.VISIBLE);
        cardRecommendation.setVisibility(View.GONE);

        // Generate recommendation (simulate AI processing)
        new android.os.Handler().postDelayed(() -> {
            // FIXED LINE: Just pass the 'goal' variable calculated above
            currentRecommendation = generateRecommendation(age, weight, height, conditions, goal);

            displayRecommendation(currentRecommendation);
            layoutLoading.setVisibility(View.GONE);
            cardRecommendation.setVisibility(View.VISIBLE);
        }, 1500);
    }

    private void displayRecommendation(MassageRecommendation rec) {
        txtRecommendedMode.setText(rec.mode);
        txtRecommendedLevel.setText("Level " + rec.level + " of 5");
        txtRecommendedDuration.setText(rec.duration + " minutes");
        txtRecommendedHeat.setText(rec.useHeat ? "Recommended ✓" : "Not needed");
        txtRecommendedHeat.setTextColor(getResources().getColor(
                // Use standard Android red
                rec.useHeat ? android.R.color.holo_red_dark : android.R.color.darker_gray));

        txtExplanation.setText(rec.explanation);
    }

    private void applyRecommendation() {
        if (currentRecommendation == null) return;

        // Send recommendation back to MainActivity
        android.content.Intent intent = new android.content.Intent();
        intent.putExtra("level", currentRecommendation.level);
        intent.putExtra("heat", currentRecommendation.useHeat);
        intent.putExtra("duration", currentRecommendation.duration);
        setResult(RESULT_OK, intent);
        finish();

        Toast.makeText(this, "Settings applied!", Toast.LENGTH_SHORT).show();
    }

    private void applyPreset(MassageRecommendation rec) {
        currentRecommendation = rec;
        displayRecommendation(rec);
        cardRecommendation.setVisibility(View.VISIBLE);
    }

    // Data class for recommendation
    private static class MassageRecommendation {
        int level;
        int duration;
        boolean useHeat;
        String mode;
        String explanation;

        MassageRecommendation(int level, int duration, boolean useHeat, String mode, String explanation) {
            this.level = level;
            this.duration = duration;
            this.useHeat = useHeat;
            this.mode = mode;
            this.explanation = explanation;
        }
    }
}

File: app/src/main/java/com/example/myapplication/ui/assistant/AssistantActivity.kt
package com.example.myapplication.ui.assistant

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.example.myapplication.R
import com.google.android.material.button.MaterialButton
import com.google.android.material.card.MaterialCardView
import com.google.android.material.chip.Chip
import com.google.android.material.textview.MaterialTextView
import kotlinx.coroutines.launch

class AssistantActivity : AppCompatActivity() {
    
    private val viewModel: AssistantViewModel by viewModels()
    
    // Input Fields
    private lateinit var edtAge: EditText
    private lateinit var edtWeight: EditText
    private lateinit var edtHeight: EditText
    
    // Health Condition Chips
    private lateinit var chipBackPain: Chip
    private lateinit var chipNeckPain: Chip
    private lateinit var chipStress: Chip
    private lateinit var chipFatigue: Chip
    private lateinit var chipInsomnia: Chip
    
    // Goal Chips
    private lateinit var chipRelaxation: Chip
    private lateinit var chipPainRelief: Chip
    private lateinit var chipRecovery: Chip
    private lateinit var chipEnergy: Chip
    
    // Buttons
    private lateinit var btnGetRecommendation: MaterialButton
    private lateinit var btnApplyRecommendation: MaterialButton
    
    // Loading and Result
    private lateinit var layoutLoading: MaterialCardView
    private lateinit var cardRecommendation: MaterialCardView
    
    // Result Display
    private lateinit var txtRecommendedMode: MaterialTextView
    private lateinit var txtRecommendedLevel: MaterialTextView
    private lateinit var txtRecommendedDuration: MaterialTextView
    private lateinit var txtRecommendedHeat: MaterialTextView
    private lateinit var txtExplanation: MaterialTextView
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_assistant)
        
        initViews()
        setupListeners()
        observeViewModel()
    }
    
    private fun initViews() {
        findViewById<View>(R.id.btnBack).setOnClickListener { finish() }
        
        // Input Fields
        edtAge = findViewById(R.id.edtAge)
        edtWeight = findViewById(R.id.edtWeight)
        edtHeight = findViewById(R.id.edtHeight)
        
        // Health Condition Chips
        chipBackPain = findViewById(R.id.chipBackPain)
        chipNeckPain = findViewById(R.id.chipNeckPain)
        chipStress = findViewById(R.id.chipStress)
        chipFatigue = findViewById(R.id.chipFatigue)
        chipInsomnia = findViewById(R.id.chipInsomnia)
        
        // Goal Chips
        chipRelaxation = findViewById(R.id.chipRelaxation)
        chipPainRelief = findViewById(R.id.chipPainRelief)
        chipRecovery = findViewById(R.id.chipRecovery)
        chipEnergy = findViewById(R.id.chipEnergy)
        
        // Buttons
        btnGetRecommendation = findViewById(R.id.btnGetRecommendation)
        btnApplyRecommendation = findViewById(R.id.btnApplyRecommendation)
        
        // Loading and Result
        layoutLoading = findViewById(R.id.layoutLoading)
        cardRecommendation = findViewById(R.id.cardRecommendation)
        
        // Result Display
        txtRecommendedMode = findViewById(R.id.txtRecommendedMode)
        txtRecommendedLevel = findViewById(R.id.txtRecommendedLevel)
        txtRecommendedDuration = findViewById(R.id.txtRecommendedDuration)
        txtRecommendedHeat = findViewById(R.id.txtRecommendedHeat)
        txtExplanation = findViewById(R.id.txtExplanation)
    }
    
    private fun setupListeners() {
        btnGetRecommendation.setOnClickListener {
            generateRecommendation()
        }
        
        btnApplyRecommendation.setOnClickListener {
            applyRecommendation()
        }
        
        // Quick presets
        findViewById<View>(R.id.cardMorningEnergy).setOnClickListener {
            applyPreset(AssistantViewModel.PresetType.MORNING_ENERGY)
        }
        
        findViewById<View>(R.id.cardRelaxation).setOnClickListener {
            applyPreset(AssistantViewModel.PresetType.RELAXATION)
        }
        
        findViewById<View>(R.id.cardDeepTissue).setOnClickListener {
            applyPreset(AssistantViewModel.PresetType.DEEP_TISSUE)
        }
    }
    
    private fun observeViewModel() {
        lifecycleScope.launch {
            viewModel.recommendation.collect { recommendation ->
                recommendation?.let {
                    displayRecommendation(it)
                }
            }
        }
        
        lifecycleScope.launch {
            viewModel.isLoading.collect { isLoading ->
                if (isLoading) {
                    layoutLoading.visibility = View.VISIBLE
                    cardRecommendation.visibility = View.GONE
                } else {
                    layoutLoading.visibility = View.GONE
                    if (viewModel.recommendation.value != null) {
                        cardRecommendation.visibility = View.VISIBLE
                    }
                }
            }
        }
    }
    
    private fun generateRecommendation() {
        // Validate input
        if (edtAge.text.toString().isEmpty() ||
            edtWeight.text.toString().isEmpty() ||
            edtHeight.text.toString().isEmpty()
        ) {
            Toast.makeText(this, "Please fill in all fields", Toast.LENGTH_SHORT).show()
            return
        }
        
        val age = edtAge.text.toString().toIntOrNull() ?: 0
        val weight = edtWeight.text.toString().toFloatOrNull() ?: 0f
        val height = edtHeight.text.toString().toFloatOrNull() ?: 0f
        
        if (age <= 0 || weight <= 0 || height <= 0) {
            Toast.makeText(this, "Please enter valid values", Toast.LENGTH_SHORT).show()
            return
        }
        
        // Get health conditions
        val conditions = mutableListOf<String>()
        if (chipBackPain.isChecked) conditions.add("back_pain")
        if (chipNeckPain.isChecked) conditions.add("neck_pain")
        if (chipStress.isChecked) conditions.add("stress")
        if (chipFatigue.isChecked) conditions.add("fatigue")
        if (chipInsomnia.isChecked) conditions.add("insomnia")
        
        // Get goal
        val goal = when {
            chipPainRelief.isChecked -> "pain_relief"
            chipRecovery.isChecked -> "recovery"
            chipEnergy.isChecked -> "energy"
            else -> "relaxation"
        }
        
        viewModel.generateRecommendation(age, weight, height, conditions, goal)
    }
    
    private fun applyPreset(presetType: AssistantViewModel.PresetType) {
        val recommendation = viewModel.getPreset(presetType)
        displayRecommendation(recommendation)
        cardRecommendation.visibility = View.VISIBLE
    }
    
    private fun displayRecommendation(recommendation: com.example.myapplication.data.models.MassageRecommendation) {
        txtRecommendedMode.text = recommendation.mode
        txtRecommendedLevel.text = "Level ${recommendation.level} of 5"
        txtRecommendedDuration.text = "${recommendation.duration} minutes"
        txtRecommendedHeat.text = if (recommendation.useHeat) "Recommended ✓" else "Not needed"
        txtRecommendedHeat.setTextColor(
            resources.getColor(
                if (recommendation.useHeat) android.R.color.holo_red_dark 
                else android.R.color.darker_gray,
                null
            )
        )
        txtExplanation.text = recommendation.explanation
    }
    
    private fun applyRecommendation() {
        val recommendation = viewModel.recommendation.value ?: return
        
        // Send recommendation back to MainActivity
        val intent = Intent().apply {
            putExtra("level", recommendation.level)
            putExtra("heat", recommendation.useHeat)
            putExtra("duration", recommendation.duration)
        }
        setResult(RESULT_OK, intent)
        finish()
        
        Toast.makeText(this, "Settings applied!", Toast.LENGTH_SHORT).show()
    }
}
